src/components/BlockDocumentsTable.vue

```diff
@@ -125,9 +141,26 @@
     blockTypes.value = []
   }
 
+  const selectedBlockDocuments = ref<string[]>([])
+  const selectAllBlockDocuments = (allBlockDocumentsSelected: CheckboxModel): string[] => {
+    if (allBlockDocumentsSelected) {
+      return selectedBlockDocuments.value = [...blockDocuments.value.flatMap(blockDocument => blockDocument.can.delete ? [blockDocument.id] : [])]
```

Comment:
`blockDocument.can.delete` should be what you want 👍 

I think ideally we could give better user feedback for each that isn't deletable rather than just not selecting it, but overall this is in the right direction. and we'd maybe do some of that handling server-side in a bulk delete endpoint anyway. 

might be weird that you could still select individual rows (where `row.can.delete == false`) but not in bulk... which will probably be the behavior using a selectable `<p-table >` unless you disable them.

Reply:
Yeah.. if we use the table select (as we should) it maybe makes sense to mutate the block document to have a disabled field if there's no delete permission. Or we move the check to the dlete button and add a toast if there's a permissions issue.